pkg/model/config/config.go

```diff
@@ -147,41 +153,43 @@ func (c Config) resourceAPIVersionCompatible(verType, version string) bool {
 	return true
 }
 
-// GVK contains information about scaffolded resources
-type GVK struct {
-	Group   string `json:"group,omitempty"`
-	Version string `json:"version,omitempty"`
-	Kind    string `json:"kind,omitempty"`
-
-	// CRDVersion holds the CustomResourceDefinition API version used for the GVK.
-	CRDVersion string `json:"crdVersion,omitempty"`
-	// WebhookVersion holds the {Validating,Mutating}WebhookConfiguration API version used for the GVK.
-	WebhookVersion string `json:"webhookVersion,omitempty"`
-}
-
-// isEqualTo compares it with another resource
-func (r GVK) isEqualTo(other GVK) bool {
-	return r.Group == other.Group &&
-		r.Version == other.Version &&
-		r.Kind == other.Kind
-}
-
-// merge combines fields of two GVKs that have matching group, version, and kind,
-// favoring the receiver's values.
-func (r *GVK) merge(other GVK) {
-	if r.CRDVersion == "" && other.CRDVersion != "" {
-		r.CRDVersion = other.CRDVersion
-	}
-	if r.WebhookVersion == "" && other.WebhookVersion != "" {
-		r.WebhookVersion = other.WebhookVersion
-	}
-}
-
 // Marshal returns the bytes of c.
 func (c Config) Marshal() ([]byte, error) {
 	// Ignore extra fields at first.
 	cfg := c
 	cfg.Plugins = nil
+
+	// Ignore some fields if v2.
+	if cfg.IsV2() {
+		for i := range cfg.Resources {
+			cfg.Resources[i].Domain = ""
+			cfg.Resources[i].Plural = ""
+			cfg.Resources[i].Path = ""
+			cfg.Resources[i].API = nil
+			cfg.Resources[i].Controller = false
+			cfg.Resources[i].Webhooks = nil
+		}
+	}
+
+	for i, r := range cfg.Resources {
+		// If the plural is regular, omit it.
+		if r.Plural == resource.RegularPlural(r.Kind) {
+			cfg.Resources[i].Plural = ""
+		}
+		// If the path is the default location, omit it.
+		if r.Path == resource.LocalPath(cfg.Repo, r.Group, r.Version, cfg.MultiGroup) {
+			cfg.Resources[i].Path = ""
+		}
```

Comment:
we have none testdata that is storing these values. I think we need to add them.

Reply:
- Plural: modifying this value is still an open issue: #1802 #1805
- Path: there is one case where we are storing this, `Pod/v1` in `testdata-v3-multigroup`

Reply:
So, the Plural one shows part of the context of #1802 #1805.
we have none support at all to no normalize plurals until we get this pr merged.

Reply:
You can actually set the plural for the webhook but it is bugged. So we do have a corner case that can set plurals so we need to support it but qe cant add tests because it is bugged.